myokit/_sim/cvodessim.py

```diff
@@ -510,7 +510,7 @@ def reset(self):
 
         """
         self._time = 0
-        self._state = list(self._default_state)
+        self._state = [float(expr) for expr in self._default_state]
```

Comment:
I realised that there was a bug in `set_default_state`, in that it evaluates whatever was given by the user and sets the default state to floats, rather than expr. Presumably we want to keep it as expr.

The problem was the `Model.map_to_state` function, which evaluates the state given. Do we want to keep this behavior and create another `map_to_state_expr` that preserves expressions, or update `map_to_state` so that it doesn't evaluate?

Reply:
Yikes!
It's kinda making me rethink the default stat idea. But it's really necessary for your use case that the simulation can have a modifiable constant thats used as/by an initial state, no?

Reply:
How about set_default_state tries to parse it as a sequence of numbers or expression, and if it fails it hands off to map_to_state?
... but then you'd want to re-use that so OK let's have a map_to_state_expr